backend/src/email/routes/tests/email-transactional.routes.test.ts

```diff
@@ -142,92 +201,435 @@ describe('POST /transactional/email/send', () => {
+      subject: '\n\n\n',
+      body: '<script></script>',
+    }
+    const res = await request(app)
+      .post(endpoint)
+      .set('Authorization', `Bearer ${apiKey}`)
+      .send({
+        ...validApiCall,
+        subject: invalidHtmlTagsSubjectAndBody.subject,
+        body: invalidHtmlTagsSubjectAndBody.body,
+      })
+
+    expect(res.status).toBe(400)
+    expect(mockSendEmail).not.toBeCalled()
+
+    const transactionalEmail = await EmailMessageTransactional.findOne({
+      where: { userId: userId.toString() },
+    })
+    expect(transactionalEmail).not.toBeNull()
+    expect(transactionalEmail).toMatchObject({
+      recipient: validApiCall.recipient,
+      from: validApiCall.from,
+      status: TransactionalEmailMessageStatus.Unsent,
+    })
+    expect(transactionalEmail?.params).toMatchObject({
+      // NB sanitisation only occurs at sending step, doesn't affect saving in params
+      subject: invalidHtmlTagsSubjectAndBody.subject,
+      body: invalidHtmlTagsSubjectAndBody.body,
+      from: validApiCall.from,
+      replyTo: validApiCall.reply_to,
+    })
+    expect(transactionalEmail?.errorCode).toBe(EMPTY_MESSAGE_ERROR_CODE)
+  })
+
+  test('Should throw an error if file type of attachment is not supported', async () => {
+    // not actually an invalid file type; FileExtensionService checks magic number
+    const invalidFileTypeAttachment = Buffer.alloc(1024 * 1024, '.')
+    const invalidFileTypeAttachmentName = 'invalid.exe'
+    // instead, we just mock the service to return false
+    const mockFileTypeCheck = jest
+      .spyOn(FileExtensionService, 'hasAllowedExtensions')
+      .mockResolvedValue(false)
+
+    const res = await request(app)
+      .post(endpoint)
+      .set('Authorization', `Bearer ${apiKey}`)
+      .field('recipient', validApiCall.recipient)
+      .field('subject', validApiCall.subject)
+      .field('body', validApiCall.body)
+      .field('from', validApiCall.from)
+      .field('reply_to', validApiCall.reply_to)
+      .attach(
+        'attachments',
+        invalidFileTypeAttachment,
+        invalidFileTypeAttachmentName
+      )
+
+    expect(res.status).toBe(400)
+    expect(mockSendEmail).not.toBeCalled()
+    expect(mockFileTypeCheck).toBeCalledTimes(1)
+    mockFileTypeCheck.mockClear()
+
+    const transactionalEmail = await EmailMessageTransactional.findOne({
+      where: { userId: userId.toString() },
+    })
+    expect(transactionalEmail).not.toBeNull()
+    expect(transactionalEmail).toMatchObject({
+      recipient: validApiCall.recipient,
+      from: validApiCall.from,
+      status: TransactionalEmailMessageStatus.Unsent,
+    })
+    expect(transactionalEmail?.params).toMatchObject({
+      from: validApiCall.from,
+      replyTo: validApiCall.reply_to,
+    })
+    expect(transactionalEmail?.errorCode).toBe(UNSUPPORTED_FILE_TYPE_ERROR_CODE)
+  })
+
+  test('Should throw an error if attached file is malicious', async () => {
```

Comment:
Can we be more specific about the `error` here?

Reply:
just checking, this and all the ones below are the same as [this comment](https://github.com/opengovsg/postmangovsg/pull/1846#discussion_r1005448762) right, which is to use a longer title to describe the test in full?

Reply:
yesser